src/pairinteraction_gui/page/base_page.py

```diff
@@ -87,37 +74,85 @@ def hideEvent(self, event: QHideEvent) -> None:
         super().hideEvent(event)
         self.window().dockwidget.setVisible(False)
 
-    def _thread_calculate(self) -> None:
+
+class CalculationPage(SimulationPage):
+    """Base class for all pages with a calculation button."""
+
+    plotwidget: PlotEnergies
+
+    def setupWidget(self) -> None:
+        super().setupWidget()
+
+        # Control panel below the plot
+        control_layout = QHBoxLayout()
+        calculate_button = QPushButton("Calculate")
+        calculate_button.setObjectName("Calculate")
+        calculate_button.clicked.connect(self.calculate_clicked)
+        control_layout.addWidget(calculate_button)
+
+        # export_button = QPushButton("Export")
```

Comment:
Remove commented-out code or add a TODO / FIXME to explain what should be done with the code in the future.